tests/unittest_inference.py

```diff
@@ -3974,6 +3974,71 @@ def test():
         with self.assertRaises(exceptions.AstroidTypeError):
             inferred.getitem(nodes.Const("4"))
 
+    def test_infer_arg_called_type_is_uninferable(self):
+        node = extract_node(
+            """
+        def func(type):
+            type #@
+        """
+        )
+        inferred = next(node.infer())
+        assert inferred is util.Uninferable
+
+    @pytest.mark.skipif(sys.version_info < (3, 8), reason="only fails on Python 3.9")
+    def test_infer_arg_called_object_when_used_as_index_is_uninferable(self):
+        node = extract_node(
+            """
+        def func(object):
+            ['list'][
+                object #@
+            ]
+        """
+        )
+        inferred = next(node.infer())
+        assert inferred is util.Uninferable
+
+    @pytest.mark.skipif(sys.version_info < (3, 8), reason="only fails on Python 3.9")
+    def test_infer_arg_called_type_when_used_as_index_is_uninferable(self):
+        # Only this case seems to fail
+        node = extract_node(
+            """
+        def func(type):
+            ['list'][
+                type #@
+            ]
+        """
+        )
+        inferred = next(node.infer())
+        assert not isinstance(inferred, nodes.ClassDef)  # was inferred as builtins.type
+        assert inferred is util.Uninferable
+
+    @pytest.mark.skipif(sys.version_info < (3, 8), reason="only fails on Python 3.9")
+    def test_infer_arg_called_type_when_used_as_subscript_is_uninferable(self):
+        # Only this case seems to fail
+        node = extract_node(
+            """
+        def func(type):
+            type[0] #@
+        """
+        )
+        inferred = next(node.infer())
+        assert not isinstance(inferred, nodes.ClassDef)  # was inferred as builtins.type
+        assert inferred is util.Uninferable
+
+    @pytest.mark.skipif(sys.version_info < (3, 8), reason="only fails on Python 3.9")
+    def test_infer_arg_called_type_defined_in_outer_scope_is_uninferable(self):
+        # Only this case seems to fail
```

Comment:
Should we remove this comment and replace it by a link to this MR ?

Reply:
Sounds good